settings: Scale plus button in stream and user group setting UI.

evykassirer opened the pull request:
<details>
<summary>screenshots for stream setting UI</summary>

before and after at 12px, 14px, 16px, and 20px

| before | after |
| ---- | ---- |
| ![Screen Shot 2025-01-29 at 15 58 06](https://github.com/user-attachments/assets/68c2354a-261e-42ab-800d-baa79dc8424e) | ![Screen Shot 2025-01-29 at 15 56 42](https://github.com/user-attachments/assets/142d1c2f-f739-4fc0-802d-1ee34eb67910) |
| ![Screen Shot 2025-01-29 at 15 58 15](https://github.com/user-attachments/assets/98bd1377-2e6f-4019-b6cd-5bcb281a468d) | ![Screen Shot 2025-01-29 at 15 56 15](https://github.com/user-attachments/assets/c3db9c05-7cdf-4341-8602-985950ba94fc) |
| ![Screen Shot 2025-01-29 at 15 58 29](https://github.com/user-attachments/assets/de0d6428-8540-474f-a83f-65a52d2f6b0e) | ![Screen Shot 2025-01-29 at 15 56 00](https://github.com/user-attachments/assets/d4e3e3ba-610c-474d-a6cf-8afef9639c3a) |
| ![Screen Shot 2025-01-29 at 15 58 47](https://github.com/user-attachments/assets/1286892a-c780-4cfb-9761-bd8acbc6a55e) | ![Screen Shot 2025-01-29 at 15 55 30](https://github.com/user-attachments/assets/9b3ffbda-6ae7-45af-8f61-430f15676f7b) |

dark mode (20px)

![Screen Shot 2025-01-29 at 16 00 31](https://github.com/user-attachments/assets/1bb1fdd7-8530-4bf1-8079-5da9b4b86110)


</details>

<details>
<summary>screenshots for user group settings</summary>


before and after at 12px, 14px, 16px, and 20px

| before | after |
| ---- | ---- |
| ![Screen Shot 2025-01-29 at 16 03 21](https://github.com/user-attachments/assets/d70b4c0f-f97c-4c69-91f8-bd657e4fc1f9) | ![Screen Shot 2025-01-29 at 16 02 26](https://github.com/user-attachments/assets/917b9064-684b-4186-8595-741f3b9874a5) |
| ![Screen Shot 2025-01-29 at 16 03 30](https://github.com/user-attachments/assets/6b701076-0bec-4252-92b4-8e196ed7ed23) | ![Screen Shot 2025-01-29 at 16 02 10](https://github.com/user-attachments/assets/c6d4d134-3df8-432d-ae7a-b8a2a9469f2d) |
| ![Screen Shot 2025-01-29 at 16 03 38](https://github.com/user-attachments/assets/d06839b0-078e-4fb8-be46-6705767f7800) | ![Screen Shot 2025-01-29 at 16 02 02](https://github.com/user-attachments/assets/e37ba9d0-6714-4403-b757-9c644d1be96e) |
| ![Screen Shot 2025-01-29 at 16 03 49](https://github.com/user-attachments/assets/17804d1d-f08f-4e6c-ab5d-6d775e27a682) | ![Screen Shot 2025-01-29 at 16 01 30](https://github.com/user-attachments/assets/576e1a2b-160d-4735-a84e-3429ef828e1f) |


</details>

evykassirer commented:
Pushing with a new plus icon. I can retake screenshots once we've confirmed the approximate styling for the plus icon

Here it is at 20px:

![image](https://github.com/user-attachments/assets/2d1ddef4-0661-4961-b0fe-b2ffe1f7a19b)

evykassirer commented:
Screenshots have been updated!

karlstolley commented:
Given the adjustments to `height` properties here, it's a little annoying that the buttons are larger than the tab switcher. Again, what's needed in this area is a thorough flex or grid rewrite, with an outer flex set to `stretch` so that we can get out of the business of setting heights like this.

alya commented:
I'm going to continue pushing back on doing major work here, as I don't like the design in the first place. :)

alya commented:
All the before/afters look fine to me -- super helpful to have the side-by-side.

alya commented:
@sahil839 do you have any feedback?

sahil839 commented:
It does not work as expected for smaller screens, am not sure if that is to be done as follow-up. Looks good other than this.

![image](https://github.com/user-attachments/assets/4348cb13-320d-4877-b288-a24e9e80cf82)

alya commented:
That seems like something that should be fixed here, not on a future PR.

evykassirer commented:
This is related to #29514, though it seems to get worse in this PR. What do we want to do here? Do we want two rows of buttons? Do we want to remove some of the buttons?

evykassirer commented:
Opened a discussion about this on CZO: https://chat.zulip.org/#narrow/channel/101-design/topic/settings.20header.20buttons.20narrow.20screens/near/2073654

sahil839 commented:
I think having two rows of buttons can be a fine temporary fix till we figure out the final design.

evykassirer commented:
This PR now has  #33344 as its first commit, to allow for ease in reviewing the rest of it with proper handling of narrower screens.

sahil839 commented:
"+" button for groups UI looks broken for all font sizes - 
![Screenshot from 2025-02-05 09-10-31](https://github.com/user-attachments/assets/ff67f79f-7e2a-46ff-89ee-44fe4399720b)

evykassirer commented:
Thanks for catching that, pushed a fix

karlstolley commented on web/styles/subscriptions.css:
```diff
@@ -59,26 +59,30 @@
-        position: relative;
-        top: -5px;
+        line-height: 1em;
+        display: block;
```
This is a flex item, so they display: block property here is unnecessary. I also couldn't see any difference when switching off the `line-height` here either, though if it needs to stay, make it unitless.

karlstolley replied:
Oh wait...it looks like a bunch of this gets overwritten in the next commit?

evykassirer replied:
Ah yeah, I wrote this commit before changing the icon. Maybe it makes sense to combine them into one commit now?

karlstolley replied:
Probably, yeah, if that's not too much work. It'll make the history cleaner here.

evykassirer replied:
It's not much work at all! Just squashing two commits, very simple. Done :)

evykassirer commented on web/styles/subscriptions.css:
```diff
@@ -59,26 +59,30 @@
 }
 
+#add_new_user_group {
+    height: 2em;
```
Great point, adding one now.

sahil839 commented:
Looks good to me.

alya commented:
@karlstolley Do you want to take another look and mark for integration review if all looks good?

karlstolley commented:
I think @evykassirer should merge the icon-swapping commit, as discussed above. I don't feel great about the combined media/container queries, but hopefully that's something we can clean up after a fresh design pass on settings.

karlstolley commented on web/templates/stream_settings/stream_settings_overlay.hbs:
```diff
@@ -13,7 +13,7 @@
                         {{#if can_create_streams}}
                         <button class="create_stream_button create_stream_plus_button tippy-zulip-delayed-tooltip" data-tooltip-template-id="create-new-stream-tooltip-template" data-tippy-placement="bottom">
-                            <span class="create_button_plus_sign">+</span>
+                            <i class="create_button_plus_sign zulip-icon zulip-icon-square-plus" aria-hidden="true"></i>
```
Slightly outside of scope here, but with `aria-hidden` on the icon (which is reasonable), I wonder if the button shouldn't have a translated `aria-label` of "Create new channel" or some such. Could be a follow up.

evykassirer replied:
Ah yeah probably. There are several other `zulip-icon-square-plus` icons with `aria-hidden` (which I copied these from) that would also benefit from that. This would be a great first contributor issue -- could you open an issue for that?

timabbott replied:
I feel like I want to look at whether there's a way to rework our model to avoid needing both tooltips and aria-label; probably worth some #frontend discussion.

evykassirer commented:
The prep commit was merged so I rebased to remove that, and also I squashed the other two commits together as discussed above. @karlstolley is it ready for integration review now?

karlstolley commented:
Yup! Marking as such. Thanks, @evykassirer!

timabbott commented:
Merged, thanks @evykassirer and @karlstolley!
